python/tchannel/tornado/message_factory.py

```diff
@@ -56,6 +58,30 @@ def __init__(self, remote_host=None, remote_host_port=None):
         self.in_checksum = {}
         self.out_checksum = {}
 
+    def build_raw_error_message(self, err):
+        """build protocol level error message based on Error object"""
+        message = ErrorMessage(
+            code=err.code,
+            tracing=err.Tracing(err.tracing.span_id,
+                                err.tracing.parent_span_id,
+                                err.tracing.trace_id,
+                                err.tracing.traceflags),
+            message=err.message,
+        )
+
+        return message
+
+    def build_protocol_error(self, message, message_id=None):
+        """build protocol level error message based on Error object"""
+
+        error = ProtocolError(
+            code=message.code,
+            message=message.message,
+            id=message_id,
+        )
+
+        return error
```

Comment:
Why is this a method in this class at all? The functionality is needed in connection.py only and doesn't depend on anything in instances of this class.